From 9606c3c90688df47a445a10be1b87ced4f315ba2 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Mon, 27 Aug 2018 01:42:45 +0900
Subject: [PATCH 3/8] tools/sppc: dockerfile for custom Pktgen repo

This update is to change dockerfiles of Pktgen for applying the change
of build script for using any of git repository.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 tools/sppc/build/ubuntu/pktgen/Dockerfile.16.04  | 12 +++++++-----
 tools/sppc/build/ubuntu/pktgen/Dockerfile.18.04  | 12 +++++++-----
 tools/sppc/build/ubuntu/pktgen/Dockerfile.latest | 12 +++++++-----
 3 files changed, 21 insertions(+), 15 deletions(-)

diff --git a/tools/sppc/build/ubuntu/pktgen/Dockerfile.16.04 b/tools/sppc/build/ubuntu/pktgen/Dockerfile.16.04
index e72dc74..dc2c503 100644
--- a/tools/sppc/build/ubuntu/pktgen/Dockerfile.16.04
+++ b/tools/sppc/build/ubuntu/pktgen/Dockerfile.16.04
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG pktgen_repo
 ARG pktgen_branch
+ARG pktgen_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -14,6 +15,7 @@ ENV https_proxy $https_proxy
 ENV no_proxy ${no_proxy}
 ENV RTE_SDK ${rte_sdk}
 ENV RTE_TARGET ${rte_target}
+ENV PKTGEN_DIR ${pktgen_dir}
 
 RUN apt-get update && apt-get install -y \
     git \
@@ -32,14 +34,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $pktgen_branch $pktgen_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${pktgen_branch} ${pktgen_repo}
 
 # Compile DPDK and pktgen
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/pktgen-dpdk
+WORKDIR ${home_dir}/${pktgen_dir}
 RUN make
 
 # Set working directory when container is launched
diff --git a/tools/sppc/build/ubuntu/pktgen/Dockerfile.18.04 b/tools/sppc/build/ubuntu/pktgen/Dockerfile.18.04
index 6097fbc..31632e4 100644
--- a/tools/sppc/build/ubuntu/pktgen/Dockerfile.18.04
+++ b/tools/sppc/build/ubuntu/pktgen/Dockerfile.18.04
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG pktgen_repo
 ARG pktgen_branch
+ARG pktgen_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -14,6 +15,7 @@ ENV https_proxy $https_proxy
 ENV no_proxy ${no_proxy}
 ENV RTE_SDK ${rte_sdk}
 ENV RTE_TARGET ${rte_target}
+ENV PKTGEN_DIR ${pktgen_dir}
 
 RUN apt-get update && apt-get install -y \
     git \
@@ -32,14 +34,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $pktgen_branch $pktgen_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${pktgen_branch} ${pktgen_repo}
 
 # Compile DPDK and pktgen
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/pktgen-dpdk
+WORKDIR ${home_dir}/${pktgen_dir}
 RUN make
 
 # Set working directory when container is launched
diff --git a/tools/sppc/build/ubuntu/pktgen/Dockerfile.latest b/tools/sppc/build/ubuntu/pktgen/Dockerfile.latest
index ae70368..fa4b8fb 100644
--- a/tools/sppc/build/ubuntu/pktgen/Dockerfile.latest
+++ b/tools/sppc/build/ubuntu/pktgen/Dockerfile.latest
@@ -7,6 +7,7 @@ ARG dpdk_repo
 ARG dpdk_branch
 ARG pktgen_repo
 ARG pktgen_branch
+ARG pktgen_dir
 
 ENV PATH ${rte_sdk}/${rte_target}/app:${PATH}
 ENV http_proxy ${http_proxy}
@@ -14,6 +15,7 @@ ENV https_proxy $https_proxy
 ENV no_proxy ${no_proxy}
 ENV RTE_SDK ${rte_sdk}
 ENV RTE_TARGET ${rte_target}
+ENV PKTGEN_DIR ${pktgen_dir}
 
 RUN apt-get update && apt-get install -y \
     git \
@@ -32,14 +34,14 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR $home_dir
-RUN git clone $dpdk_branch $dpdk_repo
-RUN git clone $pktgen_branch $pktgen_repo
+RUN git clone ${dpdk_branch} ${dpdk_repo}
+RUN git clone ${pktgen_branch} ${pktgen_repo}
 
 # Compile DPDK and pktgen
-WORKDIR $rte_sdk
-RUN make install T=$rte_target
+WORKDIR ${rte_sdk}
+RUN make install T=${rte_target}
 
-WORKDIR ${home_dir}/pktgen-dpdk
+WORKDIR ${home_dir}/${pktgen_dir}
 RUN make
 
 # Set working directory when container is launched
-- 
2.7.4

